### PR TITLE
Add an event to override LostSpawnIncreasePerUse (Issue #892)

### LINKED ISSUE #892: Add an event to override LostSpawnIncreasePerUse
Currently, each ability template has its own `LostSpawnIncreasePerUse` value. When the ability is activated, this value is added to the "bucket" responsible for spawning additional Lost waves. When the "bucket" is filled, a wave of Lost spawns.

This `LostSpawnIncreasePerUse` value is constant for each ability template; it does not depend on which weapon is used for the ability, or any other context. Which is how we arrive at silly things like _throwing_ a Proximity Mine "making noise", as the  `LostSpawnIncreasePerUse` is the same for all "grenades" thrown using Throw Grenade ability, or launched with Launch Grenade ability.

I propose adding an event that would allow to contextually override the amount of Lost-attracting noise generated by abilities.

### PR DESCRIPTION
**Pull request by Iridar**
Closes #892

### CODE REVIEW

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/XComGameState_Unit.uc [hunk 1]
```diff
@@ -9353,6 +9357,69 @@ function EventListenerReturn OnAbilityActivated(Object EventData, Object EventSo
...
+/// this value is added to the "bucket" responsible for spawning additional Lost waves. When the "bucket" is filled, a wave of Lost spawns.
+/// This LostSpawnIncreasePerUse value is constant for each ability template; it does not depend on which weapon is used for the ability or any other context. 
+/// The `XComGameState_Unit::OnAbilityActivated` triggers a `OverrideLostSpawnIncreasePerUse` event, 
+/// allowing mods to override the amount of Lost-attracting noise generated by abilities.
+/// ```unrealscript
+/// EventID: OverrideLostSpawnIncreasePerUse
+/// /// EventData: XComLWTuple {
+/// 	Data: [
+///       inout int LostSpawnModifier,
+/// 	  in XComGameState_Ability ActivatedAbilityState,
```
Comment (Xymanek) [inline X2WOTCCommunityHighlander/Src/XComGame/Classes/XComGameState_Unit.uc:9372]:
```suggestion
///       inout int LostSpawnModifier,
///       in XComGameState_Ability ActivatedAbilityState,
```

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/XComGameState_Unit.uc [hunk 2]
```diff
@@ -9353,6 +9357,69 @@ function EventListenerReturn OnAbilityActivated(Object EventData, Object EventSo
...
+
+	OverrideTuple = new class'XComLWTuple';
+	OverrideTuple.Id = 'OverrideLostSpawnIncreasePerUse';
+	OverrideTuple.Data.Add(2);
+	OverrideTuple.Data[0].Kind = XComLWTVInt;
+	OverrideTuple.Data[0].i = LostSpawnModifier;
+	OverrideTuple.Data[1].Kind = XComLWTVObject;
+	OverrideTuple.Data[1].o = ActivatedAbilityState;	
+
+	`XEVENTMGR.TriggerEvent('OverrideLostSpawnIncreasePerUse', OverrideTuple, self, GameState);
```
Comment (Xymanek) [inline X2WOTCCommunityHighlander/Src/XComGame/Classes/XComGameState_Unit.uc:9417]:
```suggestion
	`XEVENTMGR.TriggerEvent('OverrideLostSpawnIncreaseFromUse', OverrideTuple, self, GameState);
```

Reason: the original name was for the template (so multiple instances) while this is for a specific instance

Comment (Iridar) [reply X2WOTCCommunityHighlander/Src/XComGame/Classes/XComGameState_Unit.uc]: As you wish. I also renamed the event in other places of the code.

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/XComGameState_Unit.uc [hunk 3]
```diff
@@ -9353,6 +9357,69 @@ function EventListenerReturn OnAbilityActivated(Object EventData, Object EventSo
...
+// Start Issue #892
+/// HL-Docs: feature:OverrideLostSpawnIncreaseFromUse; issue:892; tags:tactical
+/// Normally, each ability template has its own LostSpawnIncreasePerUse value. When the ability is activated, 
+/// this value is added to the "bucket" responsible for spawning additional Lost waves. When the "bucket" is filled, a wave of Lost spawns.
+/// This LostSpawnIncreasePerUse value is constant for each ability template; it does not depend on which weapon is used for the ability or any other context. 
+/// The `XComGameState_Unit::OnAbilityActivated` triggers a `OverrideLostSpawnIncreaseFromUse` event, 
+/// allowing mods to override the amount of Lost-attracting noise generated by abilities.
+/// ```unrealscript
+/// EventID: OverrideLostSpawnIncreaseFromUse
+/// /// EventData: XComLWTuple {
```
Comment (robojumper) [inline X2WOTCCommunityHighlander/Src/XComGame/Classes/XComGameState_Unit.uc:9369]: 🙃

**Comment by robojumper** [comment]
LGTM otherwise.

**Pull request merged**